lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -433,48 +433,81 @@ class UseRegistryShadingModeImporter {
             return MObject();
         }
 
+        MPlug surfaceOutputPlug, volumeOutputPlug, displacementOutputPlug;
+
         // ComputeSurfaceSource will default to the universal render context if
         // renderContext is not found. Therefore we need to test first that the
         // render context output we are looking for really exists:
-        if (!shadeMaterial.GetSurfaceOutput(renderContext)) {
-            return MObject();
+        if (shadeMaterial.GetSurfaceOutput(renderContext)) {
+            UsdShadeShader surfaceShader = shadeMaterial.ComputeSurfaceSource(renderContext);
+            if (surfaceShader) {
+                const TfToken surfaceShaderPlugName = _context->GetSurfaceShaderPlugName();
+                if (!surfaceShaderPlugName.IsEmpty()) {
+                    surfaceOutputPlug = _GetSourcePlug(surfaceShader, UsdShadeTokens->surface);
+                }
+            }
         }
 
-        UsdShadeShader surfaceShader = shadeMaterial.ComputeSurfaceSource(renderContext);
-        if (!surfaceShader) {
-            return MObject();
+        if (shadeMaterial.GetVolumeOutput(renderContext)) {
+            UsdShadeShader volumeShader = shadeMaterial.ComputeVolumeSource(renderContext);
+            if (volumeShader) {
+                const TfToken volumeShaderPlugName = _context->GetVolumeShaderPlugName();
+                if (!volumeShaderPlugName.IsEmpty()) {
+                    volumeOutputPlug = _GetSourcePlug(volumeShader, UsdShadeTokens->volume);
+                }
+            }
         }
 
-        const TfToken surfaceShaderPlugName = _context->GetSurfaceShaderPlugName();
-        if (surfaceShaderPlugName.IsEmpty()) {
-            return MObject();
+        if (shadeMaterial.GetDisplacementOutput(renderContext)) {
+            UsdShadeShader displacementShader
+                = shadeMaterial.ComputeDisplacementSource(renderContext);
+            if (displacementShader) {
+                const TfToken displacementShaderPlugName
+                    = _context->GetDisplacementShaderPlugName();
+                if (!displacementShaderPlugName.IsEmpty()) {
+                    displacementOutputPlug
+                        = _GetSourcePlug(displacementShader, UsdShadeTokens->displacement);
+                }
+            }
         }
 
-        MPlug shaderOutputPlug = _GetSourcePlug(surfaceShader, UsdShadeTokens->surface);
-        if (shaderOutputPlug.isNull()) {
-            return MObject();
-        }
+        if (!surfaceOutputPlug.isNull() || !volumeOutputPlug.isNull()
+            || !displacementOutputPlug.isNull()) {
```

Comment:
Maybe invert this and return so we can save an indent level for everything below?:

```C++
        if(surfaceOutputPlug.isNull() && volumeOutputPlug.isNull() &&
                displacementOutputPlug.isNull()) {
            return MObject();
        }

        // Create the shading engine.
        ....
```